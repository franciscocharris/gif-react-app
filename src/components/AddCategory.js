import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) =>{
        setInputValue( e.target.value  );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 0){
            // console.log(`handleSubmit`, inputValue);
            // al llamar a una funcion de un hook, traera los valores si se trata como un callback
            // aqui se esta llenando el array de los gifs que se van a cargar
            setCategories( cat => [inputValue, ...cat] ); 
            setInputValue('');  
        }
    }

    return (
        <form onSubmit={handleSubmit} >
            {/* solo prueba */}
            <p>{ inputValue }</p>
            <input 
                type='text'
                placeholder='Buscar Gifs'
                value={inputValue}
                // el onchange se dispara cada vez que cambien el valor del input
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}