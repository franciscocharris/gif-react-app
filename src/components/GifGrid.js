import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

// para renombrar con la desestructuracion, solo es dos puntos : y el nuevo nombre
    const { data:images, loading} = useFetchGifs(category);
    
    // el useEffect sirve para ejecutar codigo con una condicion, se le pasa parametros y si es un array vacio
    // entonces significa que se ejecute una sola vez, cuamdo se renderize el componente
    // useEffect( () => {
    //     getGifs( category )
    //         // .then( imgs => setImages(imgs) );
    //         .then( setImages);
    // }, [category]);

    return (
        <>
            <h3 className='animate__animated animate__animated' ><span>{ category }</span></h3>
            {loading && <p className='animate__animated animate__flash'>Loading</p> }
            <div className='card-grid'>
                    {
                        images.map( img => 
                            <GifGridItem 
                                key={img.id} 
                                // se le hace un spread operator a la coleccion de objetos para enviar propiedades de manera
                                // individual
                                {...img}
                            />
                        )
                    }
            </div>
        </>
    )
}
