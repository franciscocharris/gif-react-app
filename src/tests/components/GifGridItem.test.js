import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en <GifGridItem/>', () => {
    const title = 'el titulo';
    const url = 'el url';
    // shallow para renderizar o tener la informacion del componente renderizado
    const wrapper = shallow( <GifGridItem  title={title}  url={url}  /> );

    // uno lo puede comparar con los snapshots
    test('debe mostrar el componente correctamente', () => {
        // con tener un componente, se debe importar react
        // console.log(wrapper);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('asegurar que el parrafo tenga el title', () => {
        const p = wrapper.find('p');
        
        expect(p.text()).toBe(title);
    });

    test('verificar que la imagen tenga los props en los lugares correctos', () => {
        const img = wrapper.find('img');
        // console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe tener la clase animate__bounceInUp', () => {
        const div = wrapper.find('.animate__bounceInUp');
        console.log(div.props().className);  
        const className = div.props().className;
        expect(className.includes('animate__bounceInUp')).toBe(true);
    })
    
    
    
    
});



