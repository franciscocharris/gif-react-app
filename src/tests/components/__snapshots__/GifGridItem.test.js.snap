// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem/> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceInUp"
>
  <img
    alt="el titulo"
    src="el url"
  />
  <p>
    el titulo
  </p>
</div>
`;
